include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_utils.h

```diff
@@ -681,6 +682,21 @@ class __static_monotonic_dispatcher<::std::integer_sequence<::std::uint16_t, _X,
     }
 };
 
+template <typename _Commutative>
+struct __is_commutative_operator : _Commutative
+{
+};
+
+template <typename _SpirvCommutative, typename _NonSpirvCommutative>
+struct __is_commutative_operator<oneapi::dpl::__internal::__commutative_matrix<_SpirvCommutative, _NonSpirvCommutative>>
+    :
+#if _ONEDPL_DETECT_SPIRV_COMPILATION
+    _SpirvCommutative
+#else
+    _NonSpirvCommutative
+#endif
```

Comment:
These preprocessor directives can be replaced in a single line using a `std::conditional`.

```suggestion
    : std::conditional_t<_ONEDPL_DETECT_SPIRV_COMPILATION, _SpirvCommutative, _NonSpirvCommutative>
```

Reply:
Thanks for the suggestion. The code has been moved around a bit to consolidate the implementation.

In the case we are not using the SYCL backend, this macro will be undefined which will result in compilation issues with `std::conditional`, so I think using the preprocessor directives is best here.